db/fetcher/fetcher_acp.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package fetcher
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/defradb/acp"
+	"github.com/sourcenetwork/defradb/db/permission"
+)
+
+// runDocReadPermissionCheck handles the checking (while fetching) if the document has read access
+// or not, according to our access logic based on weather (1) the request is permissioned,
+// (2) the collection is permissioned (has a policy), (3) acp module exists.
+func (df *DocumentFetcher) runDocReadPermissionCheck(ctx context.Context) error {
```

Comment:
suggestion: Splitting this function out to a new file makes it harder to read `DocumentFetcher` IMO, especially given that you are mutating the private state of the host object (`passedPermissionCheck`).  Please move it back into fetcher.go

Reply:
- [ ] revisit if worth doing

Reply:
- [x] This file and function is completely removed now, made sense to do it without a function while I was doing another suggestion.